### ISSUE TITLE
IOS Background mode: skip forward/back

### ISSUE THREAD

**Issue by Enragedsaturday** [doublesymmetry/react-native-track-player#1319]
**Describe the bug**
While skipping forward and/or backward while the app is backgrounded on notification player, the forward/back button will randomly gray out and loose the ability to skip. There are still tracks available behind/ahead in the queue. Opening up the app and calling skiptonext/skiptoprevious works just fine. 

**To Reproduce**
Have a tracklayer spun up with multiple tracks in queue. Background app. Use notification controls to skip forward/back a few times. Eventually a forward or backward skip button will gray out and not function.

**Environment (please complete the following information):**
ystem:
    OS: macOS 11.6
    CPU: (6) x64 Intel(R) Core(TM) i5-8500B CPU @ 3.00GHz
    Memory: 207.29 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.6 - /usr/local/bin/node
    Yarn: Not Found
    npm: 7.23.0 - /usr/local/bin/npm
    Watchman: 2021.10.18.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 29.0.2, 30.0.2
      System Images: android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7678000
    Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_292 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.1 => 0.66.1 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

What `react-native-track-player` version are you using?
v2.1.1
Are you testing on a real device or in the simulator? Which OS version are you running?
Real device iPhone 7, iPhone 13 pro max, iPad Pro 11 (2018). Issue present on all devices and on both iOS 14 and 15.

**Code**
I looked through the native code but I'm not that fluent in swift. I wasn't able to locate the offender. If I had to guess for some reason the remote player is thinking that the previous/next track is no longer available in the queue but I couldn't figure out why.

**Comment by Enragedsaturday** [comment]
Confirmed present in the example project provided in this repo. I'll be reverting to 1.2.7 for now.

**Comment by bradleyflood** [comment]
Thanks @Enragedsaturday are you able to investigate and/or propose a fix PR for this?

**Comment by Enragedsaturday** [comment]
@bradleyflood I tried to track it down but I'm a humble JS dev and couldn't sift through the swift to figure out the cause.

**Comment by bradleyflood** [comment]
@Enragedsaturday me too heh. Ok let's hope for an iOS dev. Know anyone? hah

**Comment by jspizziri** [comment]
@Enragedsaturday would you be able to test and see if this is an issue in `v2.2.0-rc3` ([here's the branch](https://github.com/doublesymmetry/react-native-track-player/tree/feature/kotlinaudio)).

**Comment by jspizziri** [comment]
I just tested this against `main` and I'm unable to reproduce this behavior. [It's possible it was fixed by this upgrade](https://github.com/doublesymmetry/react-native-track-player/commit/0dc7f86b8c18910abd72c6cc4d12951ad366d870#diff-1f29eaa9d372d77158ae80a9864525d1601dd12e80463e27264d399c1513a492L21). I did a lot of testing in different scenarios and in all cases I was seeing the behaviour I would expect. I'm going to close this issue as solved. If you're able to create a reproducible example against `main` I'm happy to reopen.
